File: problems_026to050/problem_030/solution.py
import os
import sys

PROJECT_PATH = os.path.dirname(os.path.dirname(os.path.dirname(os.path.abspath(__file__))))
sys.path.append(PROJECT_PATH)

from measurement import Stopwatch


class Solution(Stopwatch):
    def __init__(self):
        super().__init__()

    def execute(self):
        answer = 0
        for num in range(10, 354295):
            sum_of_powers = sum(int(digit)**5 for digit in str(num))
            if sum_of_powers == num:
                answer += num
        return answer

if __name__ == '__main__':
    soln = Solution()
    result = soln.execute()
    print(result)

File: problems_026to050/problem_030/solution.js
const { sum } = require('mathjs');

const solution = () => {
  let answer = 0;

  for (let num = 10; num <= 354294; num += 1) {
    const sumOfPowers = sum(String(num).split('').map((digit) => Number(digit) ** 5));

    if (sumOfPowers === num) {
      answer += num;
    }
  }

  return answer;
};

if (process.env.NODE_ENV !== 'test') {
  // eslint-disable-next-line no-console
  console.log(solution());
}

module.exports = solution;
